builtin/providers/opc/resource_ip_address_association.go

```diff
@@ -0,0 +1,151 @@
+package opc
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-oracle-terraform/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceOPCIPAddressAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOPCIPAddressAssociationCreate,
+		Read:   resourceOPCIPAddressAssociationRead,
+		Update: resourceOPCIPAddressAssociationUpdate,
+		Delete: resourceOPCIPAddressAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_address_reservation": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"vnic": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tagsOptionalSchema(),
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceOPCIPAddressAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*compute.Client).IPAddressAssociations()
+
+	input := compute.CreateIPAddressAssociationInput{
+		Name: d.Get("name").(string),
+	}
+
+	if ipAddressReservation, ok := d.GetOk("ip_address_reservation"); ok {
+		input.IPAddressReservation = ipAddressReservation.(string)
+	}
+
+	if vnic, ok := d.GetOk("vnic"); ok {
+		input.Vnic = vnic.(string)
+	}
+
+	tags := getStringList(d, "tags")
+	if len(tags) != 0 {
+		input.Tags = tags
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		input.Description = description.(string)
+	}
+
+	info, err := client.CreateIPAddressAssociation(&input)
+	if err != nil {
+		return fmt.Errorf("Error creating IP Address Association: %s", err)
+	}
+
+	d.SetId(info.Name)
+	return resourceOPCIPAddressAssociationRead(d, meta)
+}
+
+func resourceOPCIPAddressAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*compute.Client).IPAddressAssociations()
+
+	getInput := compute.GetIPAddressAssociationInput{
+		Name: d.Id(),
+	}
+	result, err := client.GetIPAddressAssociation(&getInput)
+	if err != nil {
+		// IP Address Association does not exist
+		if compute.WasNotFoundError(err) {
```

Comment:
Can the IP Address Association not exist, without an error? Should we check for a nil result outside of the error check?